inc/XamlApplication.cpp

```diff
@@ -0,0 +1,165 @@
+        auto coreWindowInterop = coreWindow.as<ICoreWindowInterop>();
+        HWND coreWindowHwnd = nullptr;
+        coreWindowInterop->get_WindowHandle(&coreWindowHwnd);
+        ::ShowWindow(coreWindowHwnd, SW_HIDE);
+      }
+    } else {
+      m_executionMode = ExecutionMode::UWP;
+    }
+  }
+
+  //void XamlApplication::Run(winrt::Microsoft::Toolkit::Win32::UI::XamlHost::DesktopWindow window) {
+  //  window.Show();
+  //  Run();
+  //}
+
+  void XamlApplication::Run() {
+    MSG msg = {};
+    while (GetMessage(&msg, nullptr, 0, 0)) {
+      TranslateMessage(&msg);
+      DispatchMessage(&msg);
+    }
+
+    return; //(int)msg.wParam;
+  }
+
+  winrt::Windows::Foundation::IClosable XamlApplication::WindowsXamlManager() const {
+    return m_windowsXamlManager;
+  }
+
+  void XamlApplication::Close() {
+    if (m_bIsClosed) {
+      return;
+    }
+
+    m_bIsClosed = true;
+
+    m_windowsXamlManager.Close();
+    m_providers.Clear();
+    m_windowsXamlManager = nullptr;
+
+    Exit();
+    {
+      MSG msg = {};
+      while (PeekMessageW(&msg, nullptr, 0, 0, PM_REMOVE)) {
+        ::DispatchMessageW(&msg);
+      }
+    }
+  }
+
+  XamlApplication::~XamlApplication() {
+    Close();
+  }
+
+  ::winrt::Windows::UI::Xaml::Markup::IXamlType XamlApplication::GetXamlType(::winrt::Windows::UI::Xaml::Interop::TypeName const& type) {
+    for (const auto& provider : m_providers) {
+      const auto xamlType = provider.GetXamlType(type);
+      if (xamlType != nullptr) {
+        return xamlType;
+      }
+    }
+
+    return nullptr;
+  }
+
+  ::winrt::Windows::UI::Xaml::Markup::IXamlType XamlApplication::GetXamlType(winrt::hstring const& fullName) {
+    for (const auto& provider : m_providers) {
+      const auto xamlType = provider.GetXamlType(fullName);
+      if (xamlType != nullptr) {
+        return xamlType;
+      }
+    }
+
+    return nullptr;
+  }
+
+  winrt::com_array<::winrt::Windows::UI::Xaml::Markup::XmlnsDefinition> XamlApplication::GetXmlnsDefinitions() {
+    std::list<::winrt::Windows::UI::Xaml::Markup::XmlnsDefinition> definitions;
+    for (const auto& provider : m_providers) {
+      auto defs = provider.GetXmlnsDefinitions();
+      for (const auto& def : defs) {
+        definitions.insert(definitions.begin(), def);
+      }
+    }
+
+    return winrt::com_array<::winrt::Windows::UI::Xaml::Markup::XmlnsDefinition>(definitions.begin(), definitions.end());
+  }
+
+  winrt::Windows::Foundation::Collections::IVector<::winrt::Windows::UI::Xaml::Markup::IXamlMetadataProvider> XamlApplication::MetadataProviders() {
+    return m_providers;
+  }
+}
+
+namespace winrt::Microsoft::Toolkit::Win32::UI::XamlHost::factory_implementation {
+  XamlApplication::XamlApplication() {
+    // Workaround a bug where twinapi.appcore.dll and threadpoolwinrt.dll gets loaded after it has been unloaded
+    // because of a call to GetActivationFactory
+    const wchar_t* preloadDlls[] = {

```

Comment:
remove this. these bugs have either long been fixed or are unnecessary in the common usage.